market_nextdoor_api/tests/views_tests/test_markets_location.py

```diff
@@ -0,0 +1,124 @@
+            "location_y": "39.75278314431321",
+            "distance": "0.13159076759891786",
+            "term": ""
+        },
+        {
+            "directory_type": "farmersmarket",
+            "directory_name": "farmers market",
+            "updatetime": "Mar 20th, 2023",
+            "listing_image": "default-farmersmarket-4-3.jpg",
+            "listing_id": "309132",
+            "listing_name": "Urban Farmers' Market",
+            "listing_desc": "null",
+            "brief_desc": "",
+            "mydesc": "",
+            "contact_name": "null",
+            "contact_email": "Email not found",
+            "contact_phone": "720-272-7467",
+            "media_website": "http://urbanmarketdenver.com/",
+            "media_facebook": "",
+            "media_twitter": "null",
+            "media_instagram": "null",
+            "media_pinterest": "null",
+            "media_youtube": "null",
+            "media_blog": "null",
+            "location_address": "16th St. Mall and Arapahoe, Skyline Park, Denver, CO, 80202",
+            "location_state": "Colorado",
+            "location_city": "Denver",
+            "location_street": "16th St. Mall and Arapahoe",
+            "location_zipcode": "80202",
+            "location_x": "-104.99566998565368",
+            "location_y": "39.747849996084284",
+            "distance": "0.299144076737284",
+            "term": ""
+        }
+      ]
+    }
+    #This utilizes the new library 'responses' to mock a response
+    responses.add(responses.GET, f'{self.usda_api_url}?apikey={os.environ.get("USDA_API_KEY")}&zip={self.zipcode}&radius={self.radius}', json=mock_response, status=200)
+
+    response = self.client.get(self.endpoint)
+    response_data = response.json()
+    #testing the response keys once parsed by the method
+    self.assertEqual(response.status_code, status.HTTP_200_OK)
+    self.assertIn('market_name', response_data[0])
+    self.assertIn('address', response_data[0])
+    self.assertIn('lat', response_data[0])
+    self.assertIn('lon', response_data[0])
+    self.assertIn('website', response_data[0])
+    self.assertIn('zipcode', response_data[0])
+    self.assertIn('phone', response_data[0])
+
+    #testing datatypes of values
+    self.assertIsInstance(response_data[0]['market_name'], str)
+    self.assertIsInstance(response_data[0]['address'], str)
+    self.assertIsInstance(response_data[0]['lat'], str)
+    self.assertIsInstance(response_data[0]['lon'], str)
+    self.assertIsInstance(response_data[0]['website'], str)
+    self.assertIsInstance(response_data[0]['zipcode'], str)
+    self.assertIsInstance(response_data[0]['phone'], str)
+
+  
+#SAD PATH TESTING
+  @responses.activate
+  #something to note about this mock API key, utilizing it in the subsequent f-string, 
+  #you can either use the mocked key itself, or the Environment variable which is USDA_API_KEY
+  #The convention stated to use the env variable as it's closests to how the real code works
```

Comment:
Nice! This is interesting! Would you have time to talk more about this on Monday during a standup?

Reply:
Absolutely

Reply:
Agreed, would love a walk through of how the Mock data is working. This looks great!